### PR TITLE
Office365 feed

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
reference: https://github.com/demisto/etc/issues/19714

## Description
O365 feed with multiselect of prototypes(services and regions)

## Screenshots
Screen Shot 2019-12-25 at 11 25 35

## Required version of Demisto
5.5

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation - later
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/6379c4ad18b0a64e093379c28e132707406bdd40/Integrations/FeedO365/FeedO365.yml)

### CODE REVIEW

File: Integrations/FeedO365/FeedO365.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        client: Client object with request
+
+    Returns:
+        Indicators.
+    """
+    indicator_type = client.indicator_type
+    indicator_type_lower = indicator_type.lower()
+    iterator = client.build_iterator()
+    indicators = []
+    for item in iterator:
```
Comment (yuvalbenshalom) [inline Integrations/FeedO365/FeedO365.py]: can't fetch indicators and get indicators use the same sub function ti get the indicators and handle the raw response?

File: Integrations/FeedO365/FeedO365.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        result = []
+        for url in self._base_url:
+            response = requests.get(
+                url=url,
+                verify=self._verify
+            )
+            try:
+                response.raise_for_status()
+                data = response.json()
+                result.extend([i for i in data if 'ips' in i or 'urls' in i])  # filter empty entries
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Integrations/FeedO365/FeedO365.py]: These are the indicators that this feed has.

File: Integrations/FeedO365/FeedO365.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    iterator = client.build_iterator()
+    indicator_type_lower = indicator_type.lower()
+    indicators = []
+    raw_response = []
+    limit = int(demisto.args().get('limit')) if 'limit' in demisto.args() else 100
+    iterator = [i for i in iterator if indicator_type_lower in i]  # filter indicator_type specific entries
+    iterator = iterator[:limit]
+    for item in iterator:
+        values = item.get(indicator_type_lower)
+        raw_json = {'type': indicator_type[:-1]}
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Integrations/FeedO365/FeedO365.py]: SInce ips -> ip, urls -> url

File: Integrations/FeedO365/FeedO365.py (hunk 1 above)
Comment (bakatzir) [reply Integrations/FeedO365/FeedO365.py]: If i want to limit the results prior to iterating all of them, i think it's less confusing to split it.

File: Integrations/FeedO365/FeedO365.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            try:
+                response.raise_for_status()
+                data = response.json()
+                result.extend([i for i in data if 'ips' in i or 'urls' in i])  # filter empty entries
+            except ValueError as err:
+                raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {err}')
+        return result
+
+
+def batch_indicators(sequence, batch_size=1) -> List:
```
[the remark this answers is not in the document]
Comment (DeanArbel) [reply Integrations/FeedO365/FeedO365.py]: I think we've reached the point where we should move this function to `commonServerPython`. All feed integrations should use the batch function.

**Comment by bakatzir** [comment]
@yuvalbenshalom Waiting for the Indicators enums in commonserverpython instead of hardcoding the return values in `check_indicator_type` before merging.

**Comment by yuvalbenshalom** [comment]
@bakatzir 
1. if check_indicator_type will be ready, than add it, but it is not a must.
2. please check the tests
3. I am assigning @kirbles19

**Comment by kirbles19** [comment]
@bakatzir Done. I sent a couple of questions in Slack, might have to update one or two things.

**Pull request merged**